testslide/__init__.py

```diff
@@ -50,6 +74,21 @@ def dot_lookup(thing, comp, import_path):
     return thing
 
 
+async def _async_ensure_no_leaked_tasks(coro):
+    before_example_tasks = get_all_tasks()
+    result = await coro
+    after_example_tasks = get_all_tasks()
+    new_still_running_tasks = set(after_example_tasks) - set(before_example_tasks)
+    if new_still_running_tasks:
+        tasks_str = "\n".join(str(task) for task in new_still_running_tasks)
+        raise RuntimeError(
```

Comment:
Other asyncio health issues raise their own `BaseException` classes. Reason for this, is to avoid the scenario where this exception is raised under code being tested such as:

```
try:
  do_clowny_async_thing()
except Exception:
  ...
```

and the tested code swallows the exception, tests go green, but are actually broken.

Can we have something like `class LeftOverActiveTasks(BaseException):` instead?